Flaky TestCassandraTokenSplitManager.testNonEmptyTable

https://github.com/prestosql/presto/pull/2466/checks?check_run_id=383456821

```
2020-01-10T14:27:00.5754506Z 2020-01-10T08:27:00.574-0600 INFO Life cycle stopped.
2020-01-10T14:27:01.5326820Z 2020-01-10T08:27:01.532-0600 SEVERE [Control connection] Cannot connect to any host, scheduling retry in 8000 milliseconds
2020-01-10T14:27:02.0624640Z 2020-01-10T08:27:02.061-0600 SEVERE [Control connection] Cannot connect to any host, scheduling retry in 8000 milliseconds
2020-01-10T14:27:02.0836453Z 2020-01-10T08:27:02.082-0600 SEVERE [Control connection] Cannot connect to any host, scheduling retry in 8000 milliseconds
2020-01-10T14:27:02.0837561Z 2020-01-10T08:27:02.082-0600 SEVERE [Control connection] Cannot connect to any host, scheduling retry in 8000 milliseconds
2020-01-10T14:27:02.2608629Z 2020-01-10T08:27:02.257-0600 SEVERE [Control connection] Cannot connect to any host, scheduling retry in 10000 milliseconds
2020-01-10T14:27:02.2609855Z 2020-01-10T08:27:02.258-0600 INFO Size estimates for the table test_cassandra_token_split_manager_keyspace.empty_table have been refreshed successfully: [SizeEstimate{partitionsCount=0}]
2020-01-10T14:27:02.3242789Z 2020-01-10T08:27:02.322-0600 SEVERE [Control connection] Cannot connect to any host, scheduling retry in 10000 milliseconds
2020-01-10T14:27:02.3477655Z 2020-01-10T08:27:02.346-0600 SEVERE [Control connection] Cannot connect to any host, scheduling retry in 10000 milliseconds
2020-01-10T14:27:06.4715833Z 2020-01-10T08:27:06.470-0600 INFO Size estimates for the table test_cassandra_token_split_manager_keyspace.partition_count_override_table have been refreshed successfully: [SizeEstimate{partitionsCount=0}]
2020-01-10T14:27:06.4798150Z 2020-01-10T08:27:06.479-0600 INFO Using native clock to generate timestamps.
2020-01-10T14:27:06.4806329Z 2020-01-10T08:27:06.480-0600 INFO Using native clock to generate timestamps.
2020-01-10T14:27:06.6251991Z 2020-01-10T08:27:06.623-0600 INFO Size estimates for the table test_cassandra_token_split_manager_keyspace.non_empty_table have been refreshed successfully: [SizeEstimate{partitionsCount=1152}]
2020-01-10T14:27:06.7433606Z 2020-01-10T08:27:06.742-0600 SEVERE [Control connection] Cannot connect to any host, scheduling retry in 1000 milliseconds
2020-01-10T14:27:07.5308975Z [ERROR] Tests run: 482, Failures: 1, Errors: 0, Skipped: 10, Time elapsed: 810.949 s <<< FAILURE! - in TestSuite
2020-01-10T14:27:07.5318702Z [ERROR] testNonEmptyTable(io.prestosql.plugin.cassandra.TestCassandraTokenSplitManager)  Time elapsed: 0.139 s  <<< FAILURE!
2020-01-10T14:27:07.5323650Z java.lang.AssertionError: expected [10] but found [11]
2020-01-10T14:27:07.5327552Z 	at org.testng.Assert.fail(Assert.java:94)
2020-01-10T14:27:07.5331370Z 	at org.testng.Assert.failNotEquals(Assert.java:513)
2020-01-10T14:27:07.5335975Z 	at org.testng.Assert.assertEqualsImpl(Assert.java:135)
2020-01-10T14:27:07.5340013Z 	at org.testng.Assert.assertEquals(Assert.java:116)
2020-01-10T14:27:07.5343667Z 	at org.testng.Assert.assertEquals(Assert.java:389)
2020-01-10T14:27:07.5348674Z 	at org.testng.Assert.assertEquals(Assert.java:399)
2020-01-10T14:27:07.5357638Z 	at io.prestosql.plugin.cassandra.TestCassandraTokenSplitManager.testNonEmptyTable(TestCassandraTokenSplitManager.java:99)
2020-01-10T14:27:07.5362625Z 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-01-10T14:27:07.5368866Z 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-01-10T14:27:07.5375198Z 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-01-10T14:27:07.5379238Z 	at java.lang.reflect.Method.invoke(Method.java:498)
2020-01-10T14:27:07.5385849Z 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
2020-01-10T14:27:07.5390779Z 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
2020-01-10T14:27:07.5396263Z 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
2020-01-10T14:27:07.5401315Z 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
2020-01-10T14:27:07.5407400Z 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
2020-01-10T14:27:07.5412156Z 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
2020-01-10T14:27:07.5419441Z 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2020-01-10T14:27:07.5425944Z 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2020-01-10T14:27:07.5429593Z 	at java.lang.Thread.run(Thread.java:748)
2020-01-10T14:27:07.5430732Z 
2020-01-10T14:27:07.9372975Z [INFO] 
2020-01-10T14:27:07.9373556Z [INFO] Results:
2020-01-10T14:27:07.9373926Z [INFO] 
2020-01-10T14:27:07.9375081Z [ERROR] Failures: 
2020-01-10T14:27:07.9375518Z [ERROR]   TestCassandraTokenSplitManager.testNonEmptyTable:99 expected [10] but found [11]
```

https://github.com/prestosql/presto/actions/runs/64744616

```
[ERROR] Failures: 
[ERROR]   TestCassandraTokenSplitManager.testNonEmptyTable:100 
Expected size:<10> but was:<11> in:
<[TokenSplit{startToken=7546395302881180206, endToken=-9223372036854775808, hosts=[localhost/127.0.0.1:32788]},
    TokenSplit{startToken=-7546395302881180206, endToken=-5869418568907584604, hosts=[localhost/127.0.0.1:32788]},
    TokenSplit{startToken=-838488366986797799, endToken=838488366986797802, hosts=[localhost/127.0.0.1:32788]},
    TokenSplit{startToken=-5869418568907584604, endToken=-4192441834933989002, hosts=[localhost/127.0.0.1:32788]},
    TokenSplit{startToken=-4192441834933989002, endToken=-2515465100960393400, hosts=[localhost/127.0.0.1:32788]},
    TokenSplit{startToken=-2515465100960393400, endToken=-838488366986797799, hosts=[localhost/127.0.0.1:32788]},
    TokenSplit{startToken=838488366986797802, endToken=2515465100960393403, hosts=[localhost/127.0.0.1:32788]},
    TokenSplit{startToken=2515465100960393403, endToken=4192441834933989004, hosts=[localhost/127.0.0.1:32788]},
    TokenSplit{startToken=4192441834933989004, endToken=5869418568907584605, hosts=[localhost/127.0.0.1:32788]},
    TokenSplit{startToken=5869418568907584605, endToken=7546395302881180206, hosts=[localhost/127.0.0.1:32788]},
    TokenSplit{startToken=-9223372036854775808, endToken=-7546395302881180206, hosts=[localhost/127.0.0.1:32788]}]>
```

--- a/presto-cassandra/src/test/java/io/prestosql/plugin/cassandra/TestCassandraTokenSplitManager.java
+++ b/presto-cassandra/src/test/java/io/prestosql/plugin/cassandra/TestCassandraTokenSplitManager.java
@@ -97,7 +97,9 @@ public class TestCassandraTokenSplitManager
         }
         server.refreshSizeEstimates(KEYSPACE, tableName);
         List<TokenSplit> splits = splitManager.getSplits(KEYSPACE, tableName, Optional.empty());
-        assertThat(splits).hasSize(PARTITION_COUNT / SPLIT_SIZE);
+        int expectedTokenSplitSize = PARTITION_COUNT / SPLIT_SIZE;
+        // Use hasSizeBetween because Cassandra server may overestimate the size
+        assertThat(splits).hasSizeBetween(expectedTokenSplitSize, expectedTokenSplitSize + 1);
         session.execute(format("DROP TABLE %s.%s", KEYSPACE, tableName));
     }
 }

